testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/cli/JbossCliClientJarJPMSSettingsTestCase.java

```diff
@@ -0,0 +1,130 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.as.test.manualmode.management.cli;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.PropertyPermission;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import org.jboss.as.test.integration.management.cli.CliProcessWrapper;
+import org.jboss.as.test.manualmode.management.cli.jpms.JPMSTestActivator;
+import org.jboss.as.test.shared.PermissionUtils;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.msc.service.ServiceActivator;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.impl.base.exporter.zip.ZipExporterImpl;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Launches a non-modular CLI client and deploys an application that requires specific JPMS settings. This test case is used to
+ * verify that jboss-cli-client.jar is able to run properly without requiring the user to add JVM modular settings when it
+ * launches an embedded server.
+ */
+public class JbossCliClientJarJPMSSettingsTestCase {
+    private static final Path SERVER_HOME = Paths.get(TestSuiteEnvironment.getJBossHome());
+    private static CliProcessWrapper cli;
+    private static File cliTestAppJar;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        cli = new CliProcessWrapper(false).addCliArgument("--no-color-output").addCliArgument("--timeout=30000");
+        cli.executeInteractive();
+        cliTestAppJar = createTestArchive("jpms-test-app.jar");
+    }
+
+    @AfterClass
+    public static void cleanup() throws Exception {
+        cli.destroyProcess();
+        Files.deleteIfExists(cliTestAppJar.toPath());
+    }
+
+    @Test
+    public void testManifestEntries() throws Exception {
+        JarFile clientJar = new JarFile(SERVER_HOME.resolve("bin").resolve("client").resolve("jboss-cli-client.jar").toString());
+        Manifest manifest = clientJar.getManifest();
+        Attributes mainAttribs = manifest.getMainAttributes();
+        String addOpens = mainAttribs.getValue("Add-Opens");
+        Assert.assertTrue("Unexpected value for Manifest Add-Opens. Current: " + addOpens,
+                addOpens.contains("java.base/java.util"));
+        Assert.assertTrue("Unexpected value for Manifest Add-Opens. Current: " + addOpens,
+                addOpens.contains("java.base/java.lang.invoke"));
```

Comment:
These can be a lot more extensive, i.e. what the jar contains now is:

Add-Exports: java.desktop/sun.awt java.naming/com.sun.jndi.ldap java.nam
 ing/com.sun.jndi.url.ldap java.naming/com.sun.jndi.url.ldaps
Add-Opens: java.base/java.lang java.base/java.lang.invoke java.base/java
 .lang.reflect java.base/java.io java.base/java.security java.base/java.
 util java.base/java.util.concurrent java.management/javax.management ja
 va.naming/javax.naming

If you pass maven prop embedding.jar.jpms.exports into surefire via -D you could parse it out and dynamically check for all of them.

Reply:
Thanks for reviewing it @bstansberry . Got it, I'll add a follow-up Jira for your suggestion, originally it was not my intention to verify all the settings are in the Jar.

The intention here was just to check if the required settings used by the test were satisfied to ensure we are not getting a false negative. The test case is only verifying a subset of required settings by executing something that requires them.